ext/nio4r/org/nio4r/ByteBuffer.java

```diff
@@ -0,0 +1,283 @@
+package org.nio4r;
+
+import org.jruby.*;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.javasupport.JavaUtil;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+
+/*
+created by Upekshej
+ */
+public class ByteBuffer extends RubyObject {
+
+    private java.nio.ByteBuffer byteBuffer;
+    private String currentWritePath = "";
+    private String currentReadPath = "";
+
+    private FileChannel currentWriteFileChannel;
+    private FileOutputStream fileOutputStream;
+
+    private FileInputStream currentReadChannel;
+    private FileChannel inChannel;
+
+    public ByteBuffer(final Ruby ruby, RubyClass rubyClass) {
+        super(ruby, rubyClass);
+    }
+
+    @JRubyMethod
+    public IRubyObject initialize(ThreadContext context, IRubyObject value, IRubyObject offset, IRubyObject length) {
+        Ruby ruby = context.getRuntime();
+        if (value == ruby.getNil())
+            throw new IllegalArgumentException();
```

Comment:
you probably want to have the exception type consistent across implementations, currently it is a different kind (based on backend used) : this could be improved to do `rb_raise(rb_eTypeError, "not a valid input")` like in MRI : `throw new runtime.newTypeError("not a valid input")`